wagtail/admin/templates/wagtailadmin/shared/pagination_nav.html

```diff
@@ -29,3 +71,15 @@
         </li>
     </ul>
 </nav>
+
+
+{% block extra_js %}
+    <script>
+        $("#page-size-select").on("change", function() {
+            const selectedValue = $(this).val();
+            if (selectedValue) {
+                window.location.href = selectedValue;
+            }
+        })
+    </script>
+{% endblock %}
```

Comment:
We don’t want to add any extra `<script>` tags to templates. This should most likely be done with our Stimulus controllers.

Reply:
We should be able to use `ActionController` (see `redirect`) or potentially `LinkController` for this behaviour. Some enhancements may be needed for the controllers, please be sure to include unit tests for these.

Another approach could be `SwapController`, to patch in the DOM inside the list, but that may not be what we want.

Without digging to much into this code, my guess is that `ActionController` will be the simplest, the `redirect` method already reads the url from the element's value.

Something like this should work without any JS changes.

```
<select id="page-size-select" data-controller="w-action" data-action="change->w-action#redirect">
```


Note: I removed aria-label as we would be best to provide a proper accessible `label` element.